dali/error_handling.h

```diff
@@ -244,6 +244,15 @@ inline dali::string GetStacktrace() {
 }
 #endif  // DALI_USE_STACKTRACE && DALI_DEBUG
 
+#define DALI_WARN(str)                                              \
+  do {                                                              \
+    dali::string file = __FILE__;                                   \
```

Comment:
Maybe common part between DALI_WARN and DALI_FAIL could be extracted into one macro?

Reply:
This is mostly for debugging and I don't actually think DALI_WARN will be in the final PR.